### ISSUE TITLE
Support for AWS managed Cassandra (aka AWS MCS, Amazon Keyspaces)

### ISSUE THREAD

**Issue by dahateb** [jaegertracing/jaeger#2294]
When trying to connect with the current Jaeger version to AWS MCS, the jaeger-collector connection fails with the following Error message:
> Consistency level ANY is not supported for this operation. Supported consistency levels are: ONE, LOCAL_QUORUM, LOCAL_ONE

This happens even with the commandline flag --cassandra.consistency=LOCAL_QUORUM

**To Reproduce**
1. run jaeger collector with AWS MCS configured (cassandra based  configuration with tls)

**Expected behavior**
jaeger-collector is connecting with AWS MCS

**Screenshots**

**Version (please complete the following information):**
 - OS: Amazon Linux (EKS 1.14)
 - Jaeger version: 1.18
 - Deployment: Kubernetes

deployment.yaml

```

apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-jaeger-collector
  labels:
    helm.sh/chart: jaeger-0.32.0
    app.kubernetes.io/name: jaeger
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "1.18.0"
    app.kubernetes.io/managed-by: Tiller
    app.kubernetes.io/component: collector
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: jaeger
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: collector
  strategy:
    type: Recreate
  template:
    metadata:
      annotations:
        checksum/config-env: fa599e48dfc1f40406f1b54ab64978c6cff37b3b96238cd1249f7cfdec30b956
      labels:
        app.kubernetes.io/name: jaeger
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/component: collector
    spec:
      securityContext:
        {}
        
      serviceAccountName: release-name-jaeger-collector
      containers:
      - name: release-name-jaeger-collector
        securityContext:
          {}
          
        image: jaegertracing/jaeger-collector:1.18.0
        imagePullPolicy: IfNotPresent
        args:
          - --cassandra.consistency=LOCAL_QUORUM
          
        env:
          - name: COLLECTOR_ZIPKIN_HTTP_PORT
            value: "9411"
          - name: SPAN_STORAGE_TYPE
            value: cassandra
          - name: CASSANDRA_SERVERS
            value: cassandra.eu-central-1.amazonaws.com
          - name: CASSANDRA_PORT
            value: "9142"
          
          - name: CASSANDRA_KEYSPACE
            value: XXXX
          - name: CASSANDRA_USERNAME
            value: XXXX
          - name: CASSANDRA_PASSWORD
            valueFrom:
              secretKeyRef:
                name: release-name-jaeger-cassandra
                key: password
          - name: "CASSANDRA_TLS"
            value: "true"
          
          - name: "CASSANDRA_TLS_SERVER_NAME"
            value: "cassandra.eu-central-1.amazonaws.com"
          
          - name: SAMPLING_STRATEGIES_FILE
            value: /etc/conf/strategies.json
        ports:
        - containerPort: 14250
          name: grpc
          protocol: TCP
        - containerPort: 14268
          name: http
          protocol: TCP
        - containerPort: 14269
          name: admin
          protocol: TCP
        - containerPort: 9411
          name: zipkin
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: admin
        livenessProbe:
          httpGet:
            path: /
            port: admin
        resources:
          {}
          
        volumeMounts:
          - name: cassandra-tls-secret
            mountPath: /root/.cassandra/ca-cert.pem
            subPath: ca-cert.pem
            readOnly: true
          - name: strategies
            mountPath: /etc/conf/
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      volumes:
        - name: cassandra-tls-secret
          secret:
            secretName: cassandra-tls-secret
        - name: strategies
          configMap:
            name: release-name-jaeger-sampling-strategies
``` 

**Additional context**
might be related to this fixed issue in gocql
https://github.com/gocql/gocql/pull/1422

**Comment by dahateb** [comment]
this was the original bug: https://github.com/gocql/gocql/issues/1420

**Comment by johanneswuerbach** [comment]
We also recently tried running Jaeger on AWS Keyspaces (MCS) and it doesn't support all APIs required by Jaeger yet.

E.g. Jaeger uses custom types https://github.com/jaegertracing/jaeger/blob/126701a4650325080f36e4a4fd8d8e6fd95023c1/plugin/storage/cassandra/schema/v003.cql.tmpl#L25

AWS Keyspaces doesn't support custom types https://docs.aws.amazon.com/keyspaces/latest/devguide/cassandra-apis.html

| API | Supported |
| - | - |
| CREATE TYPE | No |

**Comment by yurishkuro** [comment]
@johanneswuerbach thanks for the reference!  That page also says `frozen` types are not supported.

I think the solution for MCS would be to implement Cassandra backend where spans are stored as binary blobs, instead of using strongly typed schema, as we used to do in the early days of Jaeger with Cassandra 2.x. Strong schema only creates runtime overhead without any additional benefits. Strong schema would only be useful if we used Cassandra's native indexing capabilities, like SASI.

I am going to retitle this ticket to general support of AWS Keyspaces, as it's clear there are multiple issues that would need to be addressed.
